src/SecureStore.CyberArkCCP/Services/HttpClientCache.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace UiPath.Orchestrator.Extensions.SecureStores.CyberArkCCP.Services
+{
+    internal class HttpClientCache
+    {
+        private static readonly MemoryCacheOptions _cacheOptions = new MemoryCacheOptions
+        {
+            SizeLimit = 500,
+        };
+
+        private readonly TimeSpan _httpClientExpiration = TimeSpan.FromHours(6);
+
+        private readonly MemoryCache _clients = new MemoryCache(_cacheOptions);
+        private readonly IHttpClientFactory _httpClientFactory;
+        
+        public HttpClientCache(IHttpClientFactory httpClientFactory)
+        {
+            _httpClientFactory = httpClientFactory;
+        }
+
+        public bool Exists(string key) =>
+           _clients.Get(key) != null;
+
+        public HttpClient Get(string key) =>
+            _clients.Get<HttpClient>(key);
+
+        public HttpClient AddWithCertificate(string key, string certificateAuthorityThumbprint, X509Certificate2 certificate) =>
+            _clients.GetOrCreate(key, e => 
+            {
+                e.Size = 1;
+                e.SlidingExpiration = _httpClientExpiration;
+                return _httpClientFactory.CreateWithCertificate(certificate, certificateAuthorityThumbprint);
+            });
+
+        public void TryRemove(string key)
+        {
+            if (!Exists(key))
+            {
+                return;
+            }
+            TryDispose(_clients.Get<HttpClient>(key));
+            _clients.Remove(key);
```

Comment:
First remove it and then dispose. It's possible that you return it to someone else between when you get it and when you call Dispose.